cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/SecureStoreHandler.java

```diff
@@ -78,54 +68,43 @@ public void create(HttpRequest httpRequest, HttpResponder httpResponder, @PathPa
                                       " \n" + GSON.toJson(dummy));
     }
 
-    String description = secureKeyCreateRequest.getDescription();
-    String value = secureKeyCreateRequest.getData();
-    if (Strings.isNullOrEmpty(value)) {
-      throw new BadRequestException("The data field should not be empty. This is the data that will be stored " +
-                                      "securely.");
-    }
-
-    byte[] data = value.getBytes(StandardCharsets.UTF_8);
-    secureStoreManager.put(namespace, name, data, description, secureKeyCreateRequest.getProperties());
+    secureStoreService.put(secureKeyCreateRequest, secureKeyId);
     httpResponder.sendStatus(HttpResponseStatus.OK);
   }
 
   @Path("/keys/{key-name}")
   @DELETE
   public void delete(HttpRequest httpRequest, HttpResponder httpResponder, @PathParam("namespace-id") String namespace,
-                     @PathParam("key-name") String name) throws IOException {
-    secureStoreManager.delete(namespace, name);
+                     @PathParam("key-name") String name) throws Exception {
+    SecureKeyId secureKeyId = new SecureKeyId(namespace, name);
+    secureStoreService.delete(secureKeyId);
     httpResponder.sendStatus(HttpResponseStatus.OK);
   }
 
   @Path("/keys/{key-name}")
   @GET
   public void get(HttpRequest httpRequest, HttpResponder httpResponder, @PathParam("namespace-id") String namespace,
-                  @PathParam("key-name") String name) throws IOException {
-    SecureStoreData secureStoreData = secureStore.get(namespace, name);
-    String data = new String(secureStoreData.get(), StandardCharsets.UTF_8);
+                  @PathParam("key-name") String name) throws Exception {
+    SecureKeyId secureKeyId = new SecureKeyId(namespace, name);
+    String data = new String(secureStoreService.get(secureKeyId).get(), StandardCharsets.UTF_8);
```

Comment:
will ``SecureStoreService.get(secureKeyId)`` ever be expected to return ``byte []``? Otherwise just move this ``byte[]`` to ``String`` conversion to ``SecureStoreService``.,

Also, you can use ``Bytes.toString(byte[])`` to make it less verbose. It adds the ``UTF-8`` charset by default.